hooks/useInfiniteStoriesLoading.ts

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Category, Story } from '@prezly/sdk/dist/types';
+import { PaginationProps } from 'types';
+
+async function fetchStories(
+    page: number,
+    pageSize: number,
+    category?: Category,
+): Promise<{ stories: Story[] }> {
+    const result = await fetch('/api/fetch-stories', {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+            page,
+            pageSize,
+            category,
+        }),
+    });
+
+    if (result.status !== 200) {
+        const { message } = await result.json();
+        throw new Error(message);
+    }
+
+    return result.json();
+}
+
+export const useInfiniteStoriesLoading = (
+    initialStories: Story[],
+    pagination: PaginationProps,
+    category?: Category,
+) => {
+    const [displayedStories, setDisplayedStories] = useState<Story[]>(initialStories);
+    const [currentPage, setCurrentPage] = useState(1);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const { itemsTotal, pageSize, currentPage: currentQueryPage } = pagination;
+    const totalPages = Math.ceil(itemsTotal / pageSize);
+
+    // This also disabled infinite loading if query pagination is active
+    const canLoadMore = currentQueryPage === 1 && currentPage < totalPages;
+
+    useEffect(() => {
+        setDisplayedStories(initialStories);
+    }, [initialStories]);
```

Comment:
Yeah, that's the reason.
Actually, even then I don't think the effect is needed after all.
Server side props will trigger only once per page request, so the stories shouldn't change at all. Need to check that just in case.

Reply:
Checked again, the effect was useful for when I changed pages via classic pagination, but since we decided to split the two apporaches into different components, that's no longer needed. Removing the effect then.